test/PoolManager.swap.t.sol

```diff
@@ -0,0 +1,114 @@
+        // init pools
+        key_ = PoolKey(currency0, currency1, fee.amount(), fee.tickSpacing(), IHooks(address(0)));
+
+        IPoolManager.ModifyLiquidityParams memory v4LiquidityParams = IPoolManager.ModifyLiquidityParams({
+            tickLower: lowerTick,
+            tickUpper: upperTick,
+            liquidityDelta: liquidityDelta,
+            salt: 0
+        });
+
+        v4LiquidityParams = createFuzzyLiquidityParams(key_, v4LiquidityParams);
+
+        uint160 sqrtPriceX96 = createRandomSqrtPriceX96(key_, sqrtPriceX96seed);
+
+        v3Pool =
+            IUniswapV3Pool(v3Factory.createPool(Currency.unwrap(currency0), Currency.unwrap(currency1), fee.amount()));
+        v3Pool.initialize(sqrtPriceX96);
+
+        manager.initialize(key_, sqrtPriceX96, "");
+
+        // add liquidity
+        v3Pool.mint(
+            address(this),
+            v4LiquidityParams.tickLower,
+            v4LiquidityParams.tickUpper,
+            uint128(int128(v4LiquidityParams.liquidityDelta)),
+            ""
+        );
+
+        modifyLiquidityRouter.modifyLiquidity(key_, v4LiquidityParams, "");
+    }
+
+    function swap(IUniswapV3Pool pool, PoolKey memory key_, bool zeroForOne, int256 amountSpecified)
+        internal
+        returns (int256 amount0Diff, int256 amount1Diff)
+    {
+        // v3 swap
+        (int256 amount0Delta, int256 amount1Delta) =
+            pool.swap(address(this), zeroForOne, amountSpecified, zeroForOne ? MIN_PRICE_LIMIT : MAX_PRICE_LIMIT, "");
+        // v4 swap
+        IPoolManager.SwapParams memory swapParams = IPoolManager.SwapParams({
+            zeroForOne: zeroForOne,
+            amountSpecified: amountSpecified * -1, // invert because v3 and v4 swaps use inverted signs
+            sqrtPriceLimitX96: zeroForOne ? MIN_PRICE_LIMIT : MAX_PRICE_LIMIT
+        });
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+        BalanceDelta delta = swapRouter.swap(key_, swapParams, testSettings, "");
+
+        // because signs for v3 and v4 swaps are inverted, add values up to get the difference
+        amount0Diff = amount0Delta + delta.amount0();
+        amount1Diff = amount1Delta + delta.amount1();
+    }
+
+    function uniswapV3MintCallback(uint256 amount0Owed, uint256 amount1Owed, bytes calldata) external {
+        currency0.transfer(msg.sender, amount0Owed);
+        currency1.transfer(msg.sender, amount1Owed);
+    }
+
+    function uniswapV3SwapCallback(int256 amount0Delta, int256 amount1Delta, bytes calldata) external {
+        if (amount0Delta > 0) currency0.transfer(msg.sender, uint256(amount0Delta));
+        if (amount1Delta > 0) currency1.transfer(msg.sender, uint256(amount1Delta));
+    }
+}
+
+contract V3SwapTests is V3Fuzzer {
+    function test_shouldSwapEqual(int24 lowerTick, int24 upperTick, int128 liquidityDelta, int256 sqrtPriceX96seed)
```

Comment:
nit: could we call these params unsanitized or unbounded? when the fuzzer fails on inputs here I thought the test was actually failing with these inputs but we sanitize them later on so its a little misleading

Reply:
As far as I experimented, input args of external functions are abi-decoded from calldata, which are considered sanitized, since abi decoding includes sanity checks. But it's worth constructing dirty calldata and making a low-level call to see if abi decoding reverts.